File: gymassistantv2.client/src/app/domain/TrainingSession.ts
import { Exercise } from "./Exercise";
import { TrainingSet } from "./TrainingSet";

export class TrainingSession{
    public id: number;
    public name: string;
    public trainingPlanId: number;
    public trainingSets: TrainingSet[];  
    

    constructor() {
    this.id = 0;
    this.name = "";
    this.trainingPlanId = 0;
    this.trainingSets = [];
    }
}

File: DataAccess/Entities/TrainingSession.cs
using System.ComponentModel.DataAnnotations;

namespace DataAccess.Entities
{
    public class TrainingSession: BaseEntity
    {
        public required string Name { get; set; }
        [Required]
        public TrainingPlan TrainingsPlan { get; set; }
        public ICollection<TrainingSet> TrainingSets { get; set; }

    }
}